design-proposals/klip-4-produce-data.md

```diff
@@ -0,0 +1,302 @@
+still be sent, but the extractor will overwrite the data in `ROWTIME`.
+
+### Complex Datatypes
+
+Although not in scope for V1, complex data types can all be supported following precedents of 
+other SQL-like languages. For example, Google BigQuery supports constructing STRUCT using the
+[syntax](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#declaring-a-struct-type) 
+below (see link for full set of sample syntax):
+
+| Syntax                                    | Output                                              |
+| ----------------------------------------  | --------------------------------------------------- |
+| `STRUCT(1,2,3)`                           | `STRUCT<int64,int64,int64>` *                       |
+| `STRUCT(1 AS a, 'abc' AS b)`              | `STRUCT<a int64, b string>`                         |
+| `STRUCT<x int64>(5)`                      | `STRUCT<x int64>`                                   |
+| `STRUCT(5, STRUCT(5))`                    | `STRUCT<int64, STRUCT<int64>>`                      |
+
+\* we may not want to support "unnamed" structs, thought that decision can be made later
+
+## Test plan
+
+This functionality is expected to work both only in Interactive mode, and these tests must succeed:
+
+* Test that values can be serialized into JSON/AVRO/DELIMITED
+* Test that schemas are properly verified
+* Test `NULL` and table tombstones
+* Test `CREATE STREAM/TABLE` with topics that don't exist
+* Test producing data in topics that already exist
+* Test producing data to topics that are not associated with a KSQL source
+
+## Documentation Updates
+
+* The `CREATE TABLE` and `CREATE STREAM` syntax references will have an updated table properties
+section that includes the following rows:
+> ```rst
+> +=========================+======================================================================================================+
+> | KAFKA_TOPIC (required)  | The name of the Kafka topic that backs this source. The topic must either already exist in Kafka, or |
+> |                         | PARTITIONS and REPLICAS must be specified to create the topic. Command will fail if the topic exists |
+> |                         | with different partition/replica counts.                                                             |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | PARTITIONS              | The number of partitions in the backing topic. This property must be set if creating a SOURCE without|
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | REPLICAS                | The number of replicas in the backing topic. This property must be set if creating a SOURCE without  |
```

Comment:
The Kafka cluster itself as a default number of replicas for any topics created. So we can probably make this optional, using the Kafka cluster default if not supplied.